### PR TITLE
Issue #4720 Restrict channel fee updates…

### PR DESCRIPTION
**Pull request by ddustin**
for leased channels to be less than comitted rates for both individual channels as well as group sets. If any channel being updated is above committed limits, abort the fee setting.

### CODE REVIEW

**Review by cdecker — CHANGES_REQUESTED** [review]
Currently the `test_setchannelfee_all` CI test is broken (timeout).

File: lightningd/peer_control.c [hunk 1]
```diff
@@ -2353,6 +2353,32 @@ static void set_channel_fees(struct command *cmd, struct channel *channel,
...
 
+static struct command_result *check_lease_channel_commitments(struct command *cmd,
+                                                              struct channel *channel,
+                                                              u32 base, u32 ppm)
+{
+	/* if channel is leased from us, make sure we're not exceeding promised
+	* maximums */
+	if (channel->opener != LOCAL
+	    && get_block_height(cmd->ld->topology) < channel->lease_expiry) {
+		assert(channel->lease_chan_max_ppt * 1000 < channel->lease_chan_max_ppt);
```
[the remark this answers is not in the document]
Comment (ddustin) [reply lightningd/peer_control.c:2364]: The goal was to check for integer wrapping. I'll add a comment explaining that

**Review by cdecker — COMMENTED** [review]
Thanks for the fixups, sadly tests seems not to be too happy:

 - `test_setchannelfee_all`: might be hitting the restriction despite not being a leased channel, maybe the lease detection is not working?
 - `test_channel_lease_post_expiry` complains about `setfeerates` being unavailable (call introduced in the first commit, not sure why it's not failing everywhere)

Otherwise I look forward to merging this :-)

**Comment by ddustin** [comment]
> Thanks for the fixups, sadly tests seems not to be too happy:
> 
> 
> 
>  - `test_setchannelfee_all`: might be hitting the restriction despite not being a leased channel, maybe the lease detection is not working?
> 
>  - `test_channel_lease_post_expiry` complains about `setfeerates` being unavailable (call introduced in the first commit, not sure why it's not failing everywhere)
> 
> 
> 
> Otherwise I look forward to merging this :-)

Thanks for the comments!

I thought these would be easier to resolve once I got the test framework working on my computer (I use a Mac).

So I pivoted to doing that first, then I plan on polishing off this PR =)

File: lightningd/peer_control.c [hunk 2]
```diff
@@ -1608,7 +1608,6 @@ static const struct json_command listpeers_command = {
 	json_listpeers,
 	"Show current peers, if {level} is set, include logs for {id}"
 };
-/* Comment added to satisfice AUTODATA */
```
[the remark this answers is not in the document]
Comment (ddustin) [reply lightningd/peer_control.c:1611]: I believe the filename is only accessible as "" type string literal, sadly. I don't think it's normalized either.

**Comment by cdecker** [comment]
Wondering if we can bring this up to date and add it to the merge-train :-)

**Comment by ddustin** [comment]
> Wondering if we can bring this up to date and add it to the merge-train :-)

Workin on getting all my PRs merge-train ready this week! =)

**Comment by ddustin** [comment]
> Awesome, can't wait for your changes to land 🚀

This one got pretty stale 😳 -- Gonnna look at this again a little later.

**Comment by rustyrussell** [comment]
Closing this under the "too old" rule...  Feel free to fix up and reopen!

**Pull request closed without merging**